snuba/reader.py

```diff
@@ -36,11 +34,40 @@ def execute(
         raise NotImplementedError
 
 
-class NativeDriverReader(Reader):
+DATE_TYPE_RE = re.compile(r"(Nullable\()?Date\b")
+DATETIME_TYPE_RE = re.compile(r"(Nullable\()?DateTime\b")
+
+
+def transform_date_columns(result: Result) -> Result:
+    """
+    Converts timezone-naive date and datetime values returned by ClickHouse
+    into ISO 8601 formatted strings (including the UTC offset.)
+    """
```

Comment:
For what it's worth, I don't think that this is a desirable behavior but I also don't want to change the cache serialization right now either.